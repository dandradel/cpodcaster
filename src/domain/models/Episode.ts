export interface Episode {
  id: string | number
  title: string
  date: string
  duration: number
  description: string
  audioUrl: string
  episodeContentType?: string
  episodeFileExtension?: string
}
