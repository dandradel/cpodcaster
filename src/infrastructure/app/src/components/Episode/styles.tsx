import { SxProps } from '@mui/material'

export const episodeBoxStyles: SxProps = {
  flex: 1,
  justifyContent: 'left',
  boxShadow: 3,
  p: 4,
  minHeight: '',
  height: '100%',
}
export const dividerEpisodeStyle = {
  mt: 3,
}
export const audioElementStyles = {
  width: '100%',
  marginTop: '1rem',
}
