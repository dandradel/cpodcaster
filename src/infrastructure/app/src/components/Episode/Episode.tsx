import React, { ReactElement } from 'react'
import { Box, Divider, Typography } from '@mui/material'
import './Episode.styles.css'
import { audioElementStyles, dividerEpisodeStyle, episodeBoxStyles } from './utils'
import { Episode as EpisodeI } from '../../../../../domain/models/Episode'

export interface EpisodeProps {
  episode: EpisodeI | null
}

function Episode({ episode }: EpisodeProps): ReactElement {
  const { title, description, audioUrl, episodeContentType, episodeFileExtension } =
    episode || ({} as EpisodeI)

  const audioContentType: string =
    episodeContentType && episodeFileExtension
      ? `${episodeContentType}/${episodeFileExtension}`
      : 'audio/mpeg'

  return (
    <Box sx={episodeBoxStyles} data-testid='episode-wrapper-testid'>
      <Typography variant='h5' gutterBottom data-testid='episode-name-testid'>
        {title}
      </Typography>
      <Typography
        variant='subtitle2'
        fontStyle='italic'
        fontWeight='normal'
        gutterBottom
        data-testid='episode-description-testid'
      >
        {description}
      </Typography>
      <Divider sx={dividerEpisodeStyle} />
      <audio controls muted style={audioElementStyles} data-testid='audio-podcast-testid'>
        <source src={audioUrl} type={audioContentType} />
      </audio>
    </Box>
  )
}

export default Episode
