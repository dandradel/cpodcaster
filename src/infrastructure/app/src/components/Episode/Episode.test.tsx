import React from 'react'
import '@testing-library/jest-dom'
import { cleanup, renderWithProviders, screen } from '../../shared/test/test-utils'
import Episode from './Episode'
import episodeMock from '../../shared/__mocks__/episodes.json'

describe('Episode', () => {
  const episode = episodeMock[0]
  afterEach(cleanup)

  test('renders Episode', () => {
    const container = renderWithProviders(<Episode episode={episode} />)
    expect(container).toMatchSnapshot()
  })

  test('renders Episode with name', () => {
    renderWithProviders(<Episode episode={episode} />)
    const episodeName = screen.getByTestId('episode-name-testid')
    expect(episodeName).toBeInTheDocument()
  })

  test('renders Episode with description', () => {
    renderWithProviders(<Episode episode={episode} />)
    const episodeDescription = screen.getByTestId('episode-description-testid')
    expect(episodeDescription).toBeInTheDocument()
  })

  test('renders Episode with audio', () => {
    renderWithProviders(<Episode episode={episode} />)
    const episodeAudio = screen.getByTestId('audio-podcast-testid')
    expect(episodeAudio).toBeInTheDocument()
  })
})
