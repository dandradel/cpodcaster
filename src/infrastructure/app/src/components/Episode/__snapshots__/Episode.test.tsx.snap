// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode renders Episode 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-y6c8j4"
        data-testid="episode-wrapper-testid"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
          data-testid="episode-name-testid"
        >
          Episode 1
        </h5>
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-kr9wzm-MuiTypography-root"
          data-testid="episode-description-testid"
        >
          Description 1
        </h6>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-1sjh9xo-MuiDivider-root"
        />
        <audio
          controls=""
          data-testid="audio-podcast-testid"
          style="width: 100%; margin-top: 1rem;"
        >
          <source
            src="https://www.soundhelix.com/examples/mp3/SoundHelix-Song-1.mp3"
            type="string/string"
          />
        </audio>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-y6c8j4"
      data-testid="episode-wrapper-testid"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
        data-testid="episode-name-testid"
      >
        Episode 1
      </h5>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-kr9wzm-MuiTypography-root"
        data-testid="episode-description-testid"
      >
        Description 1
      </h6>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-1sjh9xo-MuiDivider-root"
      />
      <audio
        controls=""
        data-testid="audio-podcast-testid"
        style="width: 100%; margin-top: 1rem;"
      >
        <source
          src="https://www.soundhelix.com/examples/mp3/SoundHelix-Song-1.mp3"
          type="string/string"
        />
      </audio>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
